core/common/src/main/java/alluxio/util/ConfigurationUtils.java

```diff
@@ -38,73 +39,140 @@
+   * in a simple line-oriented format and is assumed to use the ISO 8859-1 character encoding; that
+   * is each byte is one Latin1 character. Characters not in Latin1, and certain special characters,
+   * are represented in keys and elements using Unicode escapes as defined in section 3.3 of
+   * <cite>The Java&trade; Language Specification</cite>. The specified stream remains open after
+   * this method returns.
+   *
+   * @param inStream the input stream
+   * @param destination the destination map to store the result
+   * @exception IOException if an error occurred when reading from the input stream
+   * @throws IllegalArgumentException if the input stream contains a malformed Unicode escape
+   *         sequence
+   */
+  public static void load(InputStream inStream, ConcurrentHashMapV8<String, String> destination)
+      throws IOException {
+    LineReader lr = new LineReader(inStream);
+    char[] convtBuf = new char[1024];
+    int limit;
+    int keyLen;
+    int valueStart;
+    char c;
+    boolean hasSep;
+    boolean precedingBackslash;
+
+    while ((limit = lr.readLine()) >= 0) {
+      c = 0;
+      keyLen = 0;
+      valueStart = limit;
+      hasSep = false;
+
+      // System.out.println("line=<" + new String(mLineBuf, 0, limit) + ">");
+      precedingBackslash = false;
+      while (keyLen < limit) {
+        c = lr.mLineBuf[keyLen];
+        // need check if escaped.
+        if ((c == '=' || c == ':') && !precedingBackslash) {
+          valueStart = keyLen + 1;
+          hasSep = true;
+          break;
+        } else if ((c == ' ' || c == '\t' || c == '\f') && !precedingBackslash) {
+          valueStart = keyLen + 1;
+          break;
+        }
+        if (c == '\\') {
+          precedingBackslash = !precedingBackslash;
+        } else {
+          precedingBackslash = false;
+        }
+        keyLen++;
+      }
+      while (valueStart < limit) {
+        c = lr.mLineBuf[valueStart];
+        if (c != ' ' && c != '\t' && c != '\f') {
+          if (!hasSep && (c == '=' || c == ':')) {
+            hasSep = true;
+          } else {
+            break;
+          }
+        }
+        valueStart++;
+      }
+      String key = loadConvert(lr.mLineBuf, 0, keyLen, convtBuf);
+      String value = loadConvert(lr.mLineBuf, valueStart, limit - valueStart, convtBuf);
+      destination.put(key, value);
+    }
+  }
+
+  /**
    * Loads properties from resource. This method will search Classpath for the properties file with
    * the given resourceName.
    *
    * @param resourceName filename of the properties file
-   * @return a set of properties on success, or null if failed
+   * @param destination the destination map to store the result
    */
-  public static Properties loadPropertiesFromResource(String resourceName) {
-    Properties properties = new Properties();
-
+  public static void loadPropertiesFromResource(String resourceName,
+      ConcurrentHashMapV8<String, String> destination) {
```

Comment:
`ConcurrentHashMapV8<String, String>` should be `Map<String, String>` to be more general